### ISSUE TITLE
Can't reproduce MNIST results using current codes

### ISSUE THREAD

**Issue by ain-soph** [locuslab/fast_adversarial#20]
I just cloned this repo and try to run codes with provided instructions. (the code is not modified.)  
Environment: `cuda 11.3, python 3.9.6, pytorch 1.9.0, torchvision 0.10.0`, installed via miniconda.

I run `python train_mnist.py --fname ./new_result.pth` to get a model,  
and then run `python evaluate_mnist.py --fname ./new_result.pth` to evaluate the robustness.  
and run `python evaluate_mnist.py --fname ./new_result.pth --attack none` to evaluate the clean accuracy.  
<img src="https://user-images.githubusercontent.com/13214530/126612564-36f2710a-4295-4532-8990-2745d258936d.png" alt="image" width="900"/>  
The result shows that `robustness=0.00%` and `accuracy=97.71%`, meaning the trained model is not robust at all.  

However, using your pretrained model in `models/fgsm.pth` brings a robust model. (`robustness=88.38%` and `accuracy=98.50%`)  
<img src="https://user-images.githubusercontent.com/13214530/126612600-85bc9221-4fd1-4396-bc9e-eeebacd592eb.png" alt="image" width="900"/>  

Could you provide any comment on how to reproduce your pretrained results?

**Comment by ain-soph** [comment]
Please ignore the UserWarning, which is caused by https://github.com/pytorch/vision/pull/4184

**Comment by latorrefabian** [comment]
I am having the same issue, did you find a solution? @ain-soph
